pkg/internal/token/serviceprincipaltokensecret.go

```diff
@@ -70,6 +70,7 @@ func (p *servicePrincipalToken) getPoPTokenWithClientSecret(
 		p.clientID,
 		p.tenantID,
 		options,
+		nil,
```

Comment:
The more I think about this, I think we should not nil this and make it explicit by instantiating `GetSwPoPKey` inside this package

Reply:
actually you should be able to invoke `GetSwPoPKey` from pop internal package directly no ?